### PR TITLE
shortest_path() example

### PR DESCRIPTION
**Pull request by NikitaSharma1**
I added more comments and made the example more clear.

### CODE REVIEW

File: networkx/algorithms/shortest_paths/generic.py [hunk 1]
```diff
@@ -98,17 +98,17 @@ def shortest_path(G, source=None, target=None, weight=None, method="dijkstra"):
     Examples
     --------
     >>> G = nx.path_graph(5)
-    >>> print(nx.shortest_path(G, source=0, target=4))
+    >>> print(nx.shortest_path(G, source=0, target=4)) # shortest path from source=0 to source=4
     [0, 1, 2, 3, 4]
     >>> p = nx.shortest_path(G, source=0)  # target not specified
-    >>> p[4]
-    [0, 1, 2, 3, 4]
+    >>> p[3] # shortest path from source=0 to source=3
```
Comment (MridulS) [inline networkx/algorithms/shortest_paths/generic.py:104]:
```suggestion
    >>> p[3] # shortest path from source=0 to target=3
```

File: networkx/algorithms/shortest_paths/generic.py [hunk 2]
```diff
@@ -98,17 +98,17 @@ def shortest_path(G, source=None, target=None, weight=None, method="dijkstra"):
...
+    [0, 1, 2, 3]
     >>> p = nx.shortest_path(G, target=4)  # source not specified
-    >>> p[0]
-    [0, 1, 2, 3, 4]
+    >>> p[1] # shortest path from source=1 to source=4
```
Comment (MridulS) [inline networkx/algorithms/shortest_paths/generic.py:107]:
```suggestion
    >>> p[1] # shortest path from source=1 to target=4
```

File: networkx/algorithms/shortest_paths/generic.py [hunk 3]
```diff
@@ -98,17 +98,17 @@ def shortest_path(G, source=None, target=None, weight=None, method="dijkstra"):
...
+    [1, 2, 3, 4]
     >>> p = nx.shortest_path(G)  # source, target not specified
-    >>> p[0][4]
-    [0, 1, 2, 3, 4]
+    >>> p[2][4] # shortest path from source=2 to source=4
```
Comment (MridulS) [inline networkx/algorithms/shortest_paths/generic.py:110]:
```suggestion
    >>> p[2][4] # shortest path from source=2 to target=4
```

File: networkx/algorithms/shortest_paths/generic.py [hunk 4]
```diff
@@ -98,17 +98,17 @@ def shortest_path(G, source=None, target=None, weight=None, method="dijkstra"):
     Examples
     --------
     >>> G = nx.path_graph(5)
-    >>> print(nx.shortest_path(G, source=0, target=4))
+    >>> print(nx.shortest_path(G, source=0, target=4)) # shortest path from source=0 to source=4
```
Comment (MridulS) [inline networkx/algorithms/shortest_paths/generic.py:101]:
```suggestion
    >>> print(nx.shortest_path(G, source=0, target=4))
```

This is probably not required :)

**Comment by NikitaSharma1** [comment]
@MridulS I have done the changes you've requested for.

**Review by MridulS — APPROVED** [review]

**Review by dschult — APPROVED** [review]
Looks good to me!
Thanks!

**Pull request merged**